### ISSUE TITLE
profile_ivalue nodes on unexecuted code paths are asserting as undefined

### ISSUE THREAD

**Issue by kevinstephano** [csarofeen/pytorch#874]
## 🐛 Bug

The inference version of the network EfficientNet is showing an issue where a code path is not executed during profiling because it is on the training path.  A `profile_ivalue` node for training boolean never gets executed and, therefore, never gets a defined type.  In `graph_fuser.cpp`, all the `profile_ivalue` nodes are gone through to put in constant guards and there is an assert for undefined nodes that trips an assert.  It looks like the right behavior might be just to not process undefined `profille_ivalue` nodes as it is okay for them to happen when they exist on unexecuted code paths, I think.

Note, this doesn't show up in a training run because Differentiable Groups chop the graph up into smaller chunks and don't present this unexecuted path to the NVFuser.

I put up a PR with a suggested fix here:
https://github.com/csarofeen/pytorch/pull/872

```
DUMP graph_fuser.cpp:1501]   %4460 : float = prim::GetAttr[name="drop_rate"](%self)
[DUMP graph_fuser.cpp:1501]   %4461 : bool = aten::gt(%4460, %9) # /opt/conda/lib/python3.8/site-packages/timm/models/efficientnet.py:393:11
[DUMP graph_fuser.cpp:1501]   %x.41 : Float(1, 1536, strides=[1536, 1], requires_grad=0, device=cuda:0) = prim::If(%4461) # /opt/conda/lib/python3.8/site-packages/timm/models/efficientnet.py:393:8
[DUMP graph_fuser.cpp:1501]     block0():
[DUMP graph_fuser.cpp:1501]       %4463 : bool = prim::GetAttr[name="training"](%self)
[DUMP graph_fuser.cpp:1501]       %4464 : bool = aten::lt(%4460, %9) # /opt/pytorch/pytorch/torch/nn/functional.py:1166:7
[DUMP graph_fuser.cpp:1501]       %4465 : bool = prim::If(%4464) # /opt/pytorch/pytorch/torch/nn/functional.py:1166:7
[DUMP graph_fuser.cpp:1501]         block0():
[DUMP graph_fuser.cpp:1501]           -> (%27)
[DUMP graph_fuser.cpp:1501]         block1():
[DUMP graph_fuser.cpp:1501]           %4466 : bool = aten::gt(%4460, %32) # /opt/pytorch/pytorch/torch/nn/functional.py:1166:18
[DUMP graph_fuser.cpp:1501]           -> (%4466)
[DUMP graph_fuser.cpp:1501]        = prim::If(%4465) # /opt/pytorch/pytorch/torch/nn/functional.py:1166:4
[DUMP graph_fuser.cpp:1501]         block0():
[DUMP graph_fuser.cpp:1501]           %4467 : str = aten::format(%33, %4460) # /opt/pytorch/pytorch/torch/nn/functional.py:1167:25
[DUMP graph_fuser.cpp:1501]            = prim::RaiseException(%4467) # /opt/pytorch/pytorch/torch/nn/functional.py:1167:8
[DUMP graph_fuser.cpp:1501]           -> ()
[DUMP graph_fuser.cpp:1501]         block1():
[DUMP graph_fuser.cpp:1501]           -> ()
[DUMP graph_fuser.cpp:1501]       %4468 : Tensor = prim::profile[profiled_type=Tensor](%x.10)
[DUMP graph_fuser.cpp:1501]       %4469 : bool = prim::profile_ivalue(%4463)
[DUMP graph_fuser.cpp:1501]       %4470 : Tensor = aten::dropout(%x.10, %4460, %4469) # /opt/pytorch/pytorch/torch/nn/functional.py:1168:60
[DUMP graph_fuser.cpp:1501]       %4471 : Tensor = prim::profile[profiled_type=Tensor](%4470)
[DUMP graph_fuser.cpp:1501]       -> (%4470)
[DUMP graph_fuser.cpp:1501]     block1():
[DUMP graph_fuser.cpp:1501]       -> (%x.10)
```

## To Reproduce

Pytorch Benchmark command that displays the issue (note you need to manually enable nvfuser in the script):
```
python run.py -d cuda -m jit -t eval --profile timm_efficientnet
```

I think I have enough information to make a shorter repro, if needed.

**Comment by jjsjann123** [comment]
Sounds like we have the graph in the block fused across with ops from other blocks. <- that is surprising.

Let me try to run it and take a look at the graph to have a better understanding there.

**Comment by kevinstephano** [comment]
You can use this set of files to build a docker containers:

https://gitlab-master.nvidia.com/kstephano/mydockerfiles/-/tree/main/pytorch_benchmark

**Comment by kevinstephano** [comment]
If we need to make a simpler repro, this is the code that generates the issue.  The dropout path is skipped with a `training` boolean.

```
    def forward(self, x):
        x = self.forward_features(x)
        x = self.global_pool(x)
        if self.drop_rate > 0.:
            x = F.dropout(x, p=self.drop_rate, training=self.training)
        return self.classifier(x)
```

**Comment by jjsjann123** [comment]
Closing this issue since the PR with fix is merged.
